Remove Element::focusDelegate()

nt1m opened the pull request:
#### 511a0e0ebba7fb2e622ee067d6ed9773dc39c838
<pre>
Remove Element::focusDelegate()
<a href="https://bugs.webkit.org/show_bug.cgi?id=233777">https://bugs.webkit.org/show_bug.cgi?id=233777</a>
&lt;rdar://86287159&gt;

Reviewed by Wenson Hsieh.

It does not match what the HTML spec calls a &quot;focus delegate&quot;. Just inline the code in the two places instead.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::focusDelegate): Deleted.
* Source/WebCore/dom/Element.h:
(WebCore::Element::isSliderThumbElement const):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::focusDelegate): Deleted.
* Source/WebCore/html/shadow/SliderThumbElement.h:
(isType):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::isFocused const):
(WebCore::RenderTheme::disabledTextColor const):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::paintSliderThumb):

Canonical link: <a href="https://commits.webkit.org/252926@main">https://commits.webkit.org/252926@main</a>
</pre>

pxlcoder commented on Source/WebCore/rendering/RenderTheme.cpp:
```diff
@@ -931,14 +932,14 @@ bool RenderTheme::isEnabled(const RenderObject& renderer) const
 
-    auto focusDelegate = downcast<Element>(*node).focusDelegate();
-    Document& document = focusDelegate->document();
+    RefPtr delegate = is<SliderThumbElement>(element) ? downcast<SliderThumbElement>(element) : element;
```
This should be the `hostInput()` when the element is a slider thumb.

nt1m replied:
Fixed, thanks

nt1m committed:
Remove Element::focusDelegate()
https://bugs.webkit.org/show_bug.cgi?id=233777
<rdar://86287159>

Reviewed by Wenson Hsieh.

It does not match what the HTML spec calls a "focus delegate". Just inline the code in the two places instead.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::focusDelegate): Deleted.
* Source/WebCore/dom/Element.h:
(WebCore::Element::isSliderThumbElement const):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::focusDelegate): Deleted.
* Source/WebCore/html/shadow/SliderThumbElement.h:
(isType):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::isFocused const):
(WebCore::RenderTheme::disabledTextColor const):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::paintSliderThumb):

Canonical link: https://commits.webkit.org/252926@main
